src/engineImpl.ts

```diff
@@ -48,16 +48,22 @@ export const engineImpl: Engine = async function (
             const maybeSpectral = await createSpectralWithConfFile(existsConfFile)
                 .catch(e => {
                     debug(`some error occurred loading conf file: ${e}`)
-                    return undefined
+
+                    return (e instanceof Error)
+                        ? e.message
+                        : "unknown error"
                 })
 
-            if (maybeSpectral)  {
+            if (maybeSpectral && typeof maybeSpectral != "string") {
                 spectral = maybeSpectral
             } else {
                 debug("couldn't create spectral with configuration...")
 
+                const someError = maybeSpectral
```

Comment:
is this variable really needed?

Reply:
I added it so the intent is clearer. Since the alternative without a variable reads weirdly like:

```
throw new Error(
                    maybeSpectral
                        ? `A configuration file was found but an error occurred trying to load it:\n${maybeSpectral}`
                        : "A configuration file was found but an error occurred trying to load it."
                )
```

but I'm ok with changing it you think it is worth it.